app/upgrades/v24/upgrades.go

```diff
@@ -23,6 +23,10 @@ func CreateUpgradeHandler(
 			return nil, err
 		}
 
+		// Now that the TWAP keys are refactored, we can delete all time indexed TWAPs
+		// since we only need the pool indexed TWAPs.
+		keepers.TwapKeeper.DeleteAllHistoricalTimeIndexedTWAPs(ctx)
+
```

Comment:
Note to selves, we may have to make these deletes smoothed over several blocks

Reply:
I think it depends on what else exists in the upgrade handler. If there are some other larger changes then yeah I think we should, otherwise itll be just like any other epoch, but less.